pinconfigs/README.md

```diff
@@ -0,0 +1,148 @@
+# GD32 pin configurations
+
+This directory contains a set of files describing valid pin configurations for
+multiple GD32 devices, for example, that `USART0_TX` signal can only be mapped
+to `PA0` and `PC8`. These configurations can be used to generate valid pin
+mappings.
+
+GD32 devices use two distinct models to configure signal multiplexing. In
+some devices, like GD32F103XX, each pin has one or more associated alternate
+function. These alternate functions can be sometimes changed using the remap
+functionality, which allows routing all signals from a peripheral to a
+different set of pins. For this reason we will name the model used by such
+devices *AFIO model*. Some other devices, like GD32F450XX, have a simpler
+and more advanced model where each pin has a fixed list of alternate functions.
+Similarly, we will name the model used by such devices *AF model*.
+
+In the following sections the expected file format is described.
+
+## Common configuration
+
+Some fields are common independently of the pin configuration model.
+The commented example below describes common fields in detail.
+
+- `model` (required): Choose between `afio` or `af`
+- `series` (required): Series name, e.g. gd32vf103
+- `variants` (required): Each variant has a different set of valid pin
+  combinations, either because package has different number of pins or because
+  devices with certain memory configurations have less or more peripherals
+  available.
+
+Example:
+
+```yaml
+model: afio
+
+series: gd32vf103
+
+variants:
+  - package: V
+    memories: [B, 8]
+  - package: R
+    memories: [B, 8]
+  # Same package as before but different memories: a variant because it has
+  # less peripherals (and so less signals) than B-8.
+  - package: R
+    memories: [6, 4]
```

Comment:
Package is not fully correct. GD define package by 2 symbols:

For instance:
GD32F403VC`T`6
GD32F403VC`H`6

GD32F403 (SoC Serie)
V (Number of pins: R 64, V 100, Z 144)
C (Flash Size)
H (Type of package: T LQFP, H BGA, U, QFN)

Reply:
Maybe the [GD32 MCU Selection Guide](http://gd32mcu.com/download/down/document_id/206/path_type/1) can help us understand GigaDevice product series.